src/crimson/osd/shard_services.cc

```diff
@@ -730,6 +730,34 @@ seastar::future<> OSDSingletonState::send_incremental_map(
   }
 }
 
-
+seastar::future<> OSDSingletonState::send_incremental_map_to_osd(
```

Comment:
Can we use/update already existing `send_incremental_map`?

Reply:
Yeah, presumably we can get the connection and pass it in?

Reply:
> Heartbeat::maybe_share_osdmap already seems to have the connection ref.

The connection ref in Heartbeat belongs to heartbeat msgr, the osdmap handling is done in OSD which is attached to cluster_msgr, so we have to send the osdmap through connections in cluster_msgr.

How about get the OSD connection in `OSDSingletonState::send_incremental_map_to_osd()` and reuse the `OSDSingletonState::send_incremental_map()` to send the osdmap? @athanatos @Matan-B

Reply:
I've just modified the code, please take a look again, thanks:-)